src/app/navigation/navigation.component.ts

```diff
@@ -20,24 +20,27 @@ export class NavigationComponent implements OnInit, OnDestroy {
   private filter$: BehaviorSubject<string>;
   private ngUnSubScribe: Subject<void>;
 
-  constructor(private router: Router, private store: Store<UserModuleState> ) {
+  constructor(private router: Router, private store: Store<UserModuleState>) {
     this.ngUnSubScribe = new Subject();
     this.topics$ = new BehaviorSubject([]);
     this.filter$ = new BehaviorSubject('');
-    this.store.pipe(select(selectTopicsFromState)).subscribe(lol => {
+    this.store.pipe(
+      select(selectTopicsFromState),
+      takeUntil(this.ngUnSubScribe),
```

Comment:
You could actually abstract this out into some base component and extend every component from it

move the subject creation part to base component and implement onDestroy there